src/merkle/tmr.rs

```diff
@@ -87,4 +182,13 @@ mod tests {
             Tmr::PRODUCT_IV,
         );
     }
+
+    #[test]
+    fn const_powers_of_2() {
+        let n = Tmr::POWERS_OF_TWO.len();
+        let types = crate::types::Type::powers_of_two(n);
```

Comment:
Let's move `Types::powers_of_two` into this test module. It is API bloat at this point.

Reply:
> Let's move `Types::powers_of_two` into this test module. It is API bloat at this point.

I'm unsure if it's purely bloat. It's convenient in a pinch (though I guess it's also very easy to reimplement, and so far every time I've used it I've eventually found a better solution..).